code/game/machinery/canister_frame.dm

```diff
@@ -0,0 +1,92 @@
+#define FRAME_TIER_0	"0"
+#define FRAME_TIER_1	"1"
+#define FRAME_TIER_2	"2"
+//Canister Frames
+/obj/structure/canister_frame
+	name = "canister frame"
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "frame_0"
+	density = TRUE
+	var/mode_frame = FRAME_TIER_0
+
+/obj/structure/canister_frame/examine(user)
+	. = ..()
+
+/obj/structure/canister_frame/deconstruct(disassembled = TRUE)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		new /obj/item/stack/sheet/metal(loc, 5)
+	qdel(src)
+
+/obj/structure/canister_frame/machine
+	name = "canister frame"
+	desc = "A frame used to build different kinds of canisters."
+
+/obj/structure/canister_frame/machine/examine(user)
+	. = ..()
+	if(mode_frame)
+		. += "<span class='notice'>This canister frame is tier [mode_frame].</span>"
+
+/obj/structure/canister_frame/machine/update_icon_state()
+	if (mode_frame == FRAME_TIER_0)
+		icon_state = "frame_0"
+	else if (mode_frame == FRAME_TIER_1)
+		icon_state = "frame_1"
+	else if (mode_frame == FRAME_TIER_2)
+		icon_state = "frame_2"
+
+/obj/structure/canister_frame/machine/attackby(obj/item/S, mob/user, params)
+	var/is_metal_sheet = istype(S, /obj/item/stack/sheet/metal)
+	var/is_plasteel_sheet = istype(S, /obj/item/stack/sheet/plasteel)
+	var/is_titanium_sheet = istype(S, /obj/item/stack/sheet/mineral/titanium)
+	var/is_bscrystal_sheet = istype(S, /obj/item/stack/sheet/bluespace_crystal)
+	var/is_plastitanium_sheet = istype(S, /obj/item/stack/sheet/mineral/plastitanium)
+	var/obj/item/stack/ST = S
+	var/CANISTER_TIER_1 = 1
+	var/CANISTER_TIER_2 = 2
+	var/CANISTER_TIER_3 = 3
+	if (ST.get_amount() < 5)
+		to_chat(user, "<span class='warning'>You need at least five sheets for that!</span>")
+		return
+	if(is_metal_sheet && mode_frame != FRAME_TIER_1 && mode_frame != FRAME_TIER_2)
+		if(do_after(user,15, target = src))
+			var/obj/machinery/portable_atmospherics/canister/tier_1 = new /obj/machinery/portable_atmospherics/canister(drop_location())
+			if (tier_1.mode != CANISTER_TIER_1)
+				tier_1.mode = CANISTER_TIER_1
+				tier_1.update_overlays()
```

Comment:
Why are you doing this check? will the mode of the new object ever not be canister_teir_1?

Reply:
in theory the new obj should be tier 1, but i guess is just for safety?

Reply:
if you want the new object to be a tier one, just set it. There shouldn't need to be safeties here.
This is what I was talking about when I said you should set the values with mode, so 
`tier_1.heat_limit = 5000` for instance